src/main/java/cubicchunks/server/CubeProviderServer.java

```diff
@@ -192,9 +192,15 @@ public boolean tick() {
         // NOTE: the return value is completely ignored
         profiler.startSection("providerTick("+cubeMap.getSize()+")");
         long i = System.currentTimeMillis();
+        int randomTickSpeed = this.world.getGameRules().getInt("randomTickSpeed");
         Random rand = this.world.rand;
         for (Cube cube : cubeMap) {
             cube.tickCubeServer(() -> System.currentTimeMillis() - i > 40, rand);
+            if (cube.isEmpty())
+                continue;
+            int randomTickCounter = randomTickSpeed;
+            while (randomTickCounter-- > 0)
+                cube.randomTick(this.world, rand);
```

Comment:
I don't know how I missed it before, but won't it also run when the config option says it should use vanilla code?